crates/sargon/src/wrapped_radix_engine_toolkit/low_level/compiled_transaction_intent.rs

```diff
@@ -0,0 +1,106 @@
+use crate::prelude::*;
+
+#[derive(
+    Serialize,
+    Deserialize,
+    Clone,
+    Debug,
+    PartialEq,
+    Eq,
+    Hash,
+    derive_more::Display,
+    derive_more::FromStr,
+)]
+pub struct CompiledTransactionIntent(pub BagOfBytes);
```

Comment:
I think we would wanna try hiding the constructor. And then move the `impl TransactionIntent` method this file (`pub fn compile(&self) -> CompiledTransactionIntent {`)

This lacks UniFFI compatability which will be needed if we are gonna use it in Interactors.

Not sure what is easiest with @GhenadieVP s branch, if it is a newtype or if it is a single fields object. but `BagOfBytes` need be private.
And make `pub fn new` non-pub, so that the only way of creating `CompiledTransactionIntent` (outside of this file) is to call `compile` on a `TransactionIntent`.

Reply:
Thanks @CyonAlexRDX did the changes you mentioned. Additionally I am planning to add uniffi support for `CompiledTransactionIntent` like this:

```
use crate::prelude::*;
use sargon::CompiledTransactionIntent as InternalCompiledTransactionIntent;

uniffi::custom_newtype!(CompiledTransactionIntent, BagOfBytes);

#[derive(Clone, PartialEq, Eq, InternalConversion)]
pub struct CompiledTransactionIntent(BagOfBytes);

#[uniffi::export]
pub fn new_compiled_transaction_intent_sample() -> TransactionIntent {
    InternalCompiledTransactionIntent::sample().into()
}

#[uniffi::export]
pub fn new_compiled_transaction_intent_sample_other() -> TransactionIntent {
    InternalCompiledTransactionIntent::sample_other().into()
}

#[uniffi::export]
pub fn compiled_transaction_intent_decompile(compiled_intent: &CompiledTransactionIntent) -> TransactionIntent {
    compiled_intent.into_internal().decompile().into()
}
```
@GhenadieVP Do you think that is sufficient? Also @CyonAlexRDX I used `custom_newtype` like we do with `Epoch `

Reply:
Hmm it seems that private fields are not accessible from uniffi with `InternalConversion`.

Reply:
Revised implementation, with the downside that the internal type has a public constructor. @CyonAlexRDX 
```
uniffi::custom_newtype!(CompiledTransactionIntent, BagOfBytes);

#[derive(Clone, PartialEq, Eq)]
pub struct CompiledTransactionIntent(BagOfBytes);

impl From<InternalCompiledTransactionIntent> for CompiledTransactionIntent {
    fn from(value: InternalCompiledTransactionIntent) -> Self {
        Self(value.bytes().into())
    }
}

impl CompiledTransactionIntent {
    pub fn into_internal(&self) -> InternalCompiledTransactionIntent {
        self.clone().into()
    }
}

impl Into<InternalCompiledTransactionIntent> for CompiledTransactionIntent {
    fn into(self) -> InternalCompiledTransactionIntent {
        InternalCompiledTransactionIntent::new(self.0.into())
    }
}

#[uniffi::export]
pub fn new_compiled_transaction_intent_sample() -> CompiledTransactionIntent {
    InternalCompiledTransactionIntent::sample().into()
}

#[uniffi::export]
pub fn new_compiled_transaction_intent_sample_other() -> CompiledTransactionIntent {
    InternalCompiledTransactionIntent::sample_other().into()
}

#[uniffi::export]
pub fn compiled_transaction_intent_decompile(compiled_intent: &CompiledTransactionIntent) -> TransactionIntent {
    compiled_intent.into_internal().decompile().into()
}
```

Reply:
I think we nailed the export by doing the following:

* The internal type's constructor is public, because it is needed by uniffi crate, but marked as `unsafe` so no-one can use it by accident.
* The uniffi type is exported as a record, and not a type alias, but with `secret_magic` field that is post processed as not public in kotlin and swift.